README.md

```diff
@@ -1,147 +1,181 @@
 ```r
-# import the promote package and define our model function
 library(promote)
-
-model.predict <- function(request) {
-  me <- request$name
-  greeting <- paste0("Hello", me, "!")
-  greeting
-}
-
-# add metadata to attach to this model version
-promote.metadata("NAME1",value1)
-promote.metadata("NAME2",value2)
-
-# specify the username, apikey and url, and then deploy
-promote.config  <- c(
-  username = "YOUR_USERNAME",
-  apikey = "YOUR_API_KEY",
-  env = "PROMOTE_URL"
-)
-
-promote.deploy("HelloWorld", confirm = FALSE)
 ```
 
-<hr>
-
-### Setting the Auth
-
-To deploy models, you'll need to add your username, API key, and URL to the `promote.config` variable
+Import your saved model object
 ```r
-promote.config <- c(
-  username = [USERNAME],
-  apikey = [APIKEY],
-  env = [URL]
-)
+# Previously saved model 'save(my_model, file = "my_model.rda")'
+load("my_model.rda")
 ```
 <hr>
 
-### `Promote`
+#### `model.predict()`
+The `model.predict` function is used to define the API endpoint for a model and is executed each time a model is called. **This is the core of the API endpoint**
 
-The `Promote` packages has 3 methods:
-- [`promote.deploy`](#promotedeploy)
-- [`promote.metadata`](#promotemetadata)
-- [`promote.library`](#promotelibrary)
+**Arguments**
+- `data`(_Data.Frame_): the data frame generated from the json sent to the deployed model
 
-### `promote.deploy()`
-
-#### Deploy a model to Alteryx Promote
+**Example**
+```r
+model.predict <- function(data) {
+  # generate predictions from the model based on the incoming dataframe
+  predict(my_model, data)
+}
+```
+<hr>
 
-The `deploy function captures `model.predict()` and the `promote.sh` file and sends them to the Promote servers
+#### <a name="testing"></a>Test Data
+It is a good practice to test the `model.predict` function as part of the deployment script to make sure it successfully produces an output. Once deployed, the `data` being input into the `model.predict` function will always be in the form of an R [data frame](https://stat.ethz.ch/R-manual/R-devel/library/base/html/data.frame.html). The incoming JSON will be converted to a data frame using the `fromJSON()` method available from either [jsonlite](https://cran.r-project.org/web/packages/jsonlite/jsonlite.pdf) or [rjson](https://cran.r-project.org/web/packages/rjson/rjson.pdf). Which library is used can be configured in the advanced model management section of the Promote App.
 
-#### Usage
+**Example**
+```r
+testdata <- '{"X1":[1,2,3],"X2":[4,5,6]}'
+model.predict(data.frame(jsonlite::fromJSON(testdata),stringsAsFactors=TRUE))
 
-`promote.deploy(name,confirm=False)
+```
+<hr>
 
-#### Arguments
-- `name`(_string_): the name of the model to deploy to Alteryx Promote
-- `confirm`(_boolean_, optional): If `TRUE`, then user will be prompted to confirm deployment
+#### `promote.library()`
+Tell the Promote servers to install a library required to run `model.predict()`
 
-#### Examples
+**Arguments**
```

Comment:
These are changing with custom package support, but for now I the github stuff doesn't actually work. It still passes the information along but Promote won't do anything with it.  Might be worth removing src, user from this for now, it's misleading. Part of the custom package support implemented this specific syntax though (plus the URL version).